code/modules/mob/living/living_defense.dm

```diff
@@ -377,6 +377,10 @@
 	else
 		stun_absorption[key] = list("end_time" = world.time + duration, "priority" = priority, "stuns_absorbed" = 0, \
 		"visible_message" = message, "self_message" = self_message, "examine_message" = examine_message)
+	addtimer(src, "remove_stun_absorption", duration)
+
+/mob/living/proc/remove_stun_absorption()
+	stun_absorption = null

```

Comment:
The main issue was that the stun absorption effect was persisting even after it expired. I'll probably set it to an empty list instead of null.

p.s. Why does it say that you posted that message a minute _from_ now?